move topic deletion to the rest server (Fixes #2329) (#2680)
confluentinc/ksql: docs/developer-guide/syntax-reference.rst

deleted, too. Topic deletion is asynchronous, and actual removal from brokers
may take some time to complete.

> NOTE: DELETE TOPIC will not necessarily work if your kafka cluster is configured

Reviewer: Suggest using the `.. note::` syntax.

> to create topics automatically with ``auto.create.topics.enable=true``. It is

Reviewer: ```suggestion
to create topics automatically with ``auto.create.topics.enable=true``. We
```

> recommended to follow-up a few minutes afterwards to ensure that the topic was

Reviewer: ```suggestion
recommend checking after few minutes to ensure that the topic was
```

> deleted, too. Topic deletion is asynchronous, and actual removal from brokers

Reviewer 2: While you're. Could you fix this line and the above? `delete the corresponding Avro schema is deleted, too`. I think the first delete word should not be there.

may take some time to complete.

> NOTE: DELETE TOPIC will not necessarily work if your kafka cluster is configured

Reviewer: `.. note::` synatax

> to create topics automatically with ``auto.create.topics.enable=true``. It is

Reviewer: ```suggestion
to create topics automatically with ``auto.create.topics.enable=true``. We
```

> recommended to follow-up a few minutes afterwards to ensure that the topic was

Reviewer: ```suggestion
recommend checking after a few minutes afterwards to ensure that the topic was
```

>   deleted.

Reviewer 2: How will the user know this did not work? is the `auto.create.topics.enable=true` flaky when deleting the topic? Why a few minutes, btw? Shouldn't this happen immediately?

Author: `auto.create.topics.enable` means that the streams task will automatically create the topic when something is published to that topic. this means that the delete will successfully delete the topic, but since we haven't stopped the streams task, if any message is published it will re-create the topic. 

there is a short delay between enqueuing the message on the command topic and terminating the streams task. in most cases, you will terminate the query before you drop it - so this will only happen if there are multiple nodes and one is lagging (hence I suggest waiting a while).

Author: Here is the illustration of a scenario it doesn't work:
```sql
1: CREATE STREAM FOO AS SELECT * FROM BAR;
2: Server A starts CSAS_FOO_1
3: Server B starts CSAS_FOO_1
4: TERMINATE CSAS_FOO_1; DROP FOO DELETE TOPIC;
5: Server A terminates CSAS_FOO_1;
6: Server A deletes topic FOO (ON REST SIDE);
7: Server A drops FOO from metastore
8: Server B re-creates topic FOO due to auto.create
9: Server B terminates CSAS_FOO_1
10: SERVER B drops FOO form metastore
```

Reviewer 2: Oh, it is a synchronization problem. Should we have some explanation about it in the syntax-reference? I understand what you say now, but new users or users who have only 1 server might not understand this behavior, right?

Author: @JimGalasyn I'd be interested to hear your thoughts. 

I'm happy adding documentation for this, but I'm worried that adding context might just confuse a user who is happy to just follow our guidance (since even describing the problem precisely is difficult). It is not an issue with just multiple servers, it is also possible that within a single server the consumer of the command topic is delayed, in which case the same scenario as above can still happen.

Reviewer: @agavra In general, if a feature works, it goes into docs, and if there's breakage, it goes into KB. In this case, we might want to call out this situation in the [Troubleshooting](https://docs.confluent.io/current/ksql/docs/troubleshoot-ksql.html) topic, if we think a significant number of people will run across it.

Reviewer 3: KB sounds sensible, with a link to the item from the docs, ... do we have a KB???? :D

Author: I didn't know we had a KB (does that stand for Knowledge Base?) 😉
